File: bench.go
package main

import (
	"fmt"
	"time"
)

func bench() {
	var number uint64
	var start = time.Now()
	for i := 0; i < 1000000000; i++ {
		number++
	}
	fmt.Println(time.Now().Sub(start).Milliseconds())
	fmt.Println(number)
}

func main() {
	bench()
}

// 800 - 2000 ms in CompilerExplorer - Go x86-64 gc 1.22.1
// ~250ms on my machine

File: bench.cpp
#include <iostream>
#include <chrono>
//#pragma GCC optimize ("-O3") // Configure opt level
using namespace std;

void bench() {
    auto start = chrono::system_clock::now();

    long long number = 0;

    for(long long i = 0; i < 1000000000; i++) number++;

    int elapsed = (int) (chrono::system_clock::now() - start).count() / 1000 / 1000; // System time to int milliseconds

    cout << "With no optimizations:" << endl;
    cout << elapsed << endl;
    cout << number << endl;
}

int main (int argc, char const *argv[])
{
    bench();

    return 0;
}

// 400-500 ms in CompilerExplorer (no optimizations) - C++ x86-64 gcc 14.2
// ~550ms on my machine
// 0 ms with optimizations